gstreamer-pbutils/src/auto/encoding_profile.rs

```diff
@@ -0,0 +1,194 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use DiscovererInfo;
+use ffi;
+use glib;
+use glib::StaticType;
+use glib::Value;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use gst;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct EncodingProfile(Object<ffi::GstEncodingProfile, ffi::GstEncodingProfileClass>);
+
+    match fn {
+        get_type => || ffi::gst_encoding_profile_get_type(),
+    }
+}
+
+impl EncodingProfile {
+    pub fn find<'a, 'b, P: Into<Option<&'a str>>, Q: Into<Option<&'b str>>>(targetname: &str, profilename: P, category: Q) -> Option<EncodingProfile> {
+        assert_initialized_main_thread!();
+        let profilename = profilename.into();
+        let profilename = profilename.to_glib_none();
+        let category = category.into();
+        let category = category.to_glib_none();
+        unsafe {
+            from_glib_full(ffi::gst_encoding_profile_find(targetname.to_glib_none().0, profilename.0, category.0))
+        }
+    }
+
+    pub fn from_discoverer(info: &DiscovererInfo) -> Option<EncodingProfile> {
+        skip_assert_initialized!();
+        unsafe {
+            from_glib_full(ffi::gst_encoding_profile_from_discoverer(info.to_glib_none().0))
+        }
+    }
+}
+
+unsafe impl Send for EncodingProfile {}
+unsafe impl Sync for EncodingProfile {}
+
+pub trait EncodingProfileExt {
+    fn copy(&self) -> EncodingProfile;
+
+    fn get_allow_dynamic_output(&self) -> bool;
+
+    fn get_description(&self) -> Option<String>;
+
+    fn get_file_extension(&self) -> Option<String>;
+
+    fn get_format(&self) -> gst::Caps;
+
+    fn get_input_caps(&self) -> gst::Caps;
+
+    fn get_name(&self) -> Option<String>;
```

Comment:
Can this really be `None`, and the description?

Reply:
yes, both can be set to NULL currently on the C side.